### PR TITLE
Remove the deprecated Helm charts and related code

### PR DESCRIPTION
**Pull request by mattlord**
## Description
This work was deprecated in Vitess 7.0: https://github.com/vitessio/vitess/issues/6439

Signed-off-by: Matt Lord <mattalord@gmail.com>

## Related Issue(s)
https://github.com/vitessio/vitess/issues?q=is%3Aissue+is%3Aopen+helm

## Checklist
- [ ] Should this PR be backported?
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

**Comment by derekperkins** [comment]
I'm fine with the helm charts getting deleted. My only hesitation is that I still use that `release.sh` file to ship release builds of various components. It should probably be switched over to Docker Hub auto-build config, but I haven't had time to do that yet.

**Comment by mattlord** [comment]
> I'm fine with the helm charts getting deleted. My only hesitation is that I still use that `release.sh` file to ship release builds of various components. It should probably be switched over to Docker Hub auto-build config, but I haven't had time to do that yet.

What would you prefer I do here in this PR? I can move it to another location or something else. 

Thanks!

**Comment by derekperkins** [comment]
If you can move it to `/docker` for now, that'd be great

File: doc/internal/ReleaseInstructions.md [hunk 1]
```diff
@@ -243,10 +243,6 @@ And finally, click on `Publish release`.
 * Schedule and publish Tweet on Vitess account.
 * Run following script to once the `base` Docker image is live. 
 
-```
-https://github.com/vitessio/vitess/blob/master/helm/release.sh
-```
-
```
Comment (derekperkins) [inline doc/internal/ReleaseInstructions.md:249]: This should stick around too, but pointing at the new location

File: doc/DockerBuild.md [hunk 2]
```diff
@@ -1,6 +1,3 @@
-By default, the [Helm Charts](https://github.com/vitessio/vitess/tree/main/helm)
-point to the `vitess/lite` image on [Docker Hub](https://hub.docker.com/u/vitess/).
-
```
Comment (derekperkins) [inline doc/DockerBuild.md:3]: Haha, I never knew this was here. The helm charts only used `lite` at the very very beginning before k8s added stateful sets. The completed charts always used the individual binaries.

**Comment by mattlord** [comment]
Thanks, @derekperkins ! I believe that I've addressed your comments in https://github.com/vitessio/vitess/pull/8868/commits/80d1f41bdba03cf421f25aa27f0b7036756142a7

**Review by derekperkins — APPROVED** [review]
You were great while you lasted helm. So Long, and Thanks for All the Fish

**Pull request merged**